source_test.yaml

```diff
@@ -9,6 +9,7 @@
   extension: '.json'
   db_prefix: 'ALBA-'
   ignore_git: False
+  human_link: 'https://errata.almalinux.org/{{ bug.ecosystem[1].split(":")[1] }}/{{ bug.id().replace(":", "-", 1) }}.html'
```

Comment:
Hm, this introduces a dependency on ndb/our Datastore entity format in a way that isn't immediately obvious. if we change `Bug` in the future, we'd have to remember to update this.

Can we instead just support simpler string replacements here?

i.e. we can define "{{BASE_ECOSYSTEM}}", "{{BUG_ID}}" as strings that can be used here in the yaml. Our source updating implementation can just do the replacement on the fly. 

This way, all the assumptions about the bug/Datastore entity structure are made in code, which can also be more easily tested.

Reply:
> if we change `Bug` in the future, we'd have to remember to update this.

Is adding some Big Fat Comments in models.py sufficient to address this concern?

> we can define "{{BASE_ECOSYSTEM}}", "{{BUG_ID}}" as strings that can be used here in the yaml. Our source updating implementation can just do the replacement on the fly.

I think you're saying to pass specific kwargs into the template, rather than the whole bug? I had started doing that, and then decided to pass in the whole bug for maximum future flexibility without further changes, but happy to add those two kwargs for now instead if I'm understanding correctly?

Reply:
Yes, I think kwargs would be preferred here. The flexibility comes at a cost of testability and potentially making things a bit brittle.

Reply:
One challenge with doing this generically is the values and how they're arrived at vary by ecosystem, so it shifts ecosystem-specific logic further away from the ecosystem definitions and into the handler code. That was part of the reasoning behind not doing it this way in the first place...

Reply:
Feedback address in https://github.com/google/osv.dev/pull/2031/commits/5fd32773fae35b32b9280e8a4c8e380a4647cdea

Reply:
I believe there's a happy medium here. See my other comment!